plugins/woocommerce/tests/php/src/Internal/Admin/Settings/PaymentsRestControllerIntegrationTest.php

```diff
@@ -420,6 +529,218 @@ public function test_get_payment_providers_with_invalid_location() {
 		$this->assertEquals( 400, $response->get_status() );
 	}
 
+	/**
+	 * Test getting payment providers returns all the data.
+	 */
+	public function test_get_payment_providers_has_all_the_data() {
+		// Arrange.
+		// phpcs:ignore Generic.CodeAnalysis.UnusedFunctionParameter.Found
+		$filter_callback = fn( $caps ) => array(
+			'manage_woocommerce' => true,
+			'install_plugins'    => true,
+		);
+		add_filter( 'user_has_cap', $filter_callback );
+
+		$this->enable_core_paypal_pg();
+		$this->mock_payment_gateways();
+
+		// Act.
+		$request = new WP_REST_Request( 'GET', self::ENDPOINT . '/providers' );
+		$request->set_param( 'location', 'US' );
+		$response = $this->server->dispatch( $request );
+
+		// Assert.
+		$this->assertEquals( 200, $response->get_status() );
```

Comment:
Minor nitpick: Could we replace all instances of `assertEquals` with `assertSame` for type safety?

Reply:
I can do that, but we need to be careful as sometimes `assertEquals` is what we want.